charmhelpers/contrib/openstack/templates/openstack_https_frontend

```diff
@@ -22,6 +22,9 @@ Listen {{ ext_port }}
     ProxyPassReverse / http://localhost:{{ int }}/
     ProxyPreserveHost on
     RequestHeader set X-Forwarded-Proto "https"
+    {% if haproxy_enable_proxyv2 -%}
+    RemoteIPProxyProtocol On
```

Comment:
did you consider setting `RemoteIPHeader X-Forwarded-For` as well? .... otherwise how Apache2 knows what ip address to use?

https://httpd.apache.org/docs/2.4/mod/mod_remoteip.html#remoteipheader

Reply:
Hello, yes proxyv2 protocol provides the same function and safety.
https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt

but please let me know if you have any advice for the header.

Thanks.

Reply:
ack.